cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx

```diff
@@ -1022,32 +1022,89 @@ export class ToolsControlComponent extends React.PureComponent<Props, State> {
+                if (functionAttribute.input_type === 'number') {
+                    return !Number.isNaN(value);
+                }
+
+                return ['select', 'radio', 'text'].includes(functionAttribute.input_type);
+            }
+
+            switch (functionAttribute.input_type) {
+                case 'number':
+                    return ['select', 'radio', 'text'].includes(dbAttribute.input_type) && !Number.isNaN(value);
+
+                case 'text':
+                    return ['select', 'radio'].includes(dbAttribute.input_type) && value.split(' ').length === 1;
+
+                case 'select':
+                    return ['text', 'radio'].includes(dbAttribute.input_type);
+
+                case 'radio':
+                    return ['text', 'select'].includes(dbAttribute.input_type);
+
+                default:
+                    return false;
+            }
+        }
+
         return (
             <DetectorRunner
                 withCleanup={false}
                 models={detectors}
                 labels={jobInstance.labels}
                 dimension={jobInstance.dimension}
-                runInference={async (model: Model, body: object) => {
+                runInference={async (model: Model, body: DetectorMappingModel) => {
                     try {
                         this.setState({ mode: 'detection', fetching: true });
                         const result = await core.lambda.call(jobInstance.taskId, model, { ...body, frame });
                         const states = result.map(
-                            (data: any): any => new core.classes.ObjectState({
-                                shapeType: data.type,
-                                label: jobInstance.labels.filter((label: any): boolean => label.name === data.label)[0],
-                                points: data.points,
-                                objectType: ObjectType.SHAPE,
-                                frame,
-                                occluded: false,
-                                source: 'auto',
-                                attributes: (data.attributes as { name: string, value: string }[])
-                                    .reduce((mapping, attr) => {
-                                        mapping[attrsMap[data.label][attr.name]] = attr.value;
-                                        return mapping;
-                                    }, {} as Record<number, string>),
-                                zOrder: curZOrder,
-                            }),
+                            (data: any): any => {
+                                const jobLabel = jobInstance.labels
+                                    .filter((jLabel: any): boolean => jLabel.name === data.label)[0];
+                                const modelLabel = Object.entries(body.mapping)
+                                    .filter(([, value]) => value === data.label)[0][0];
```

Comment:
I think this line should be commented, It's not so obvious why we should do `[0][0]`

Reply:
Can I just rewrite the code like so?
```ts
const modelLabel = Object.entries(body.mapping)
    .filter(([, value]) => value === data.label)
    .map(([key]) => key)[0];
```